packages/gatsby-plugin-sharp/src/scheduler.js

```diff
@@ -1,88 +1,134 @@
 const _ = require(`lodash`)
 const { existsSync } = require(`fs`)
 const uuidv4 = require(`uuid/v4`)
-const queue = require(`async/queue`)
-const { processFile } = require(`./process-file`)
+const pDefer = require(`p-defer`)
+const worker = require(`./worker`)
 const { createProgress } = require(`./utils`)
 
-const toProcess = {}
-let totalJobs = 0
-const q = queue((task, callback) => {
-  task(callback)
-}, 1)
+const toProcess = new Map()
+let imagesToProcess = 0
+let imagesFinished = 0
 
 let bar
-// when the queue is empty we stop the progressbar
-q.drain = () => {
-  if (bar) {
-    bar.done()
-  }
-  totalJobs = 0
-}
-
-const getFileKey = filePath => filePath.replace(/\./g, `%2E`)
-
-const setJobToProcess = (toProcess, job, deferred) => {
-  const inputFileKey = getFileKey(job.inputPath)
-  const outputFileKey = getFileKey(job.outputPath)
-  const jobPath = `["${inputFileKey}"].["${outputFileKey}"]`
-
-  // Check if the job has already been queued. If it has, there's nothing
-  // to do, return.
-  if (_.has(toProcess, jobPath)) {
-    return { existingPromise: _.get(toProcess, `${jobPath}.deferred.promise`) }
-  }
 
-  // Check if the output file already exists so we don't redo work.
-  if (existsSync(job.outputPath)) {
-    return { existingPromise: Promise.resolve(job) }
-  }
+const executeJobs = _.throttle(
+  (pluginOptions, boundActionCreators) => {
+    toProcess.forEach(job => {
+      toProcess.delete(job.args.inputPath)
 
-  let isQueued = false
-  if (toProcess[inputFileKey]) {
-    isQueued = true
-  }
-
-  _.set(toProcess, jobPath, {
-    job: job,
-    deferred,
-  })
-
-  return { isQueued }
-}
+      try {
+        worker({
+          ...job.args,
+          pluginOptions,
+        })
+          .then(() => {
+            job.deferred.resolve()
+          })
+          .catch(job.deferred.reject)
+          .finally(() => {
+            if (bar) {
+              job.args.operations.forEach(() => {
+                bar.tick()
+              })
```

Comment:
`progress.tick` can take an amount, if also the reporter.createProgressbar has the same tick(value) then we can do:
```js
bar.tick(job.args.operations.length)
```